src/MissionManager/TransectStyleComplexItem.cc

```diff
@@ -538,6 +522,51 @@ void TransectStyleComplexItem::_adjustTransectsForTerrain(void)
         }
 
         emit lastSequenceNumberChanged(lastSequenceNumber());
+    } else {
+        //We're not following terrain - this means that all the transects are on the same altitude
+        //(relative or absolute). Nominally, this altitude should position the vehicle at
+        //cameraCalc.distanceToSurface() from the ground, but the ground is uneven and we have
+        //to pick a representative spot/reference. We could pick:
+        //(1) The higest hill under the survey polygon for safety;
+        //(2) Average/median height under the polygon to minimise the extent where we
+        //    inevitably disobey distanceToSurface.
```

Comment:
I'm envisioning trying to explain this logic to the average user (at least our users) and I think their heads would explode, (except that maybe they wouldn't because I don't think I can even figure out how to explain it in a way they'd even start to understand).  Doesn't this whole approach makes the transect widths completely unpredictable?  I feel like even an advanced user might want some control over what the altitude reference datum would be.  Maybe I'm missing something.

Reply:
I don't think you'd ever need to explain it to anyone. It will just do something reasonable even when you are not flying over a billiard table.The point is that if you are trying to hit a specific GSD for your images. You need to know the height of the surface you are surveying and then you need to fly a distance above that surface to get the specified GSD. If you are not flying over a billiard table using a relative altitude above the home position will not work since you distance above the surface will be incorrect. If you are flying over a billiard table then it works excatly like it did before.

Reply:
> If you are flying over a billiard table then it works exactly like it did before.

That is as long as you have decent quality terrain data.

Reply:
The problem with the current code comes about when the altitude of the home position differs from the surface altitude of the place you are surveying.

Reply:
With the old code if there was a big enough difference between the alt of the home position and the general alt of the survey area there was no way to set up a survey with GSD or overlap settings that would work at all.

Reply:
> Another question is what to rename the Survey/Corridor Scan "Altitude" field to. It isn't really altitude.

Maybe "Height" ?

Reply:
Yep, that all makes sense.  My worry here is that even small tweaks to the polygon might cause the "average/median" calculation to change unpredictably which might be confusing.  Maybe I'm over-thinking it a bit.